app/src/main/java/com/github/se/eventradar/ui/home/Home.kt

```diff
@@ -169,21 +172,58 @@ fun HomeScreen(
             })
       }
     } else {
-      // "Upcoming" tab content
-      // TODO: Implement upcoming events
-      Toast.makeText(context, "Upcoming events not yet available", Toast.LENGTH_SHORT).show()
-      viewModel.onTabChanged(Tab.BROWSE)
+      if (uiState.eventList.allEvents.isEmpty() && uiState.userLoggedIn) {
```

Comment:
Do you think we should still show Map View even if there are no Upcoming Events available? The map will be empty, because i think this might be what users would expect upon clicking the MapView, but we can ask others how they think about this too.

Reply:
Good point! I agree this deserves further discussion with the team to ensure it aligns with user expectations. We can consider making adjustments in the next sprint based on our findings.